libgphoto2_port/meson.build

```diff
@@ -1,34 +1,48 @@
 libusb_req         = '>= 0.1.5'
 libusb1_req        = '>= 1.0.0'
 
-ltdl_dep = cc.find_library('ltdl', required : false)
+ltdl_dep = cc.find_library(
+  'ltdl',
+  has_headers: 'ltdl.h',
+  required: false
+)
 if not ltdl_dep.found()
-  ltdl_dep = cc.find_library('libltdl', required : false)
+  ltdl_dep = cc.find_library('libltdl', has_headers: 'ltdl.h', required: true)
 endif
-libusb_dep = dependency('libusb', version: libusb_req, required: 'usb' in get_option('iolibs'))
-libusb1_dep = dependency('libusb-1.0', version: libusb1_req, required: 'libusb1' in get_option('iolibs'))
-lockdev_dep = cc.find_library('lockdev', required : false)
 
-if libusb_dep.found()
-  add_project_arguments('-DHAVE_LIBUSB=1', language: 'c')
-endif
+# This will be libusb-1.0 by default and fallback on the older libusb.
+libusb_dep = dependency('libusb-1.0', version: libusb1_req, required: 'libusb1' in get_option('iolibs'))
 
-if libusb1_dep.found()
+usb0 = false
+usb1 = false
+
+if libusb_dep.found()
+  usb1 = true
   add_project_arguments('-DHAVE_LIBUSB1=1', language: 'c')
-  if cc.has_function('libusb_strerror', dependencies: libusb1_dep, prefix: '#include<libusb.h>')
+  if cc.has_function('libusb_strerror', dependencies: libusb_dep, prefix: '#include<libusb.h>')
     add_project_arguments('-DHAVE_LIBUSB_STRERROR=1', language: 'c')
   endif
-  if cc.has_function('libusb_wrap_sys_device', dependencies: libusb1_dep, prefix: '#include<libusb.h>')
+  if cc.has_function('libusb_wrap_sys_device', dependencies: libusb_dep, prefix: '#include<libusb.h>')
     add_project_arguments('-DHAVE_LIBUSB_WRAP_SYS_DEVICE=1', language: 'c')
   endif
   if cc.compiles('''
 #include<libusb.h>
 enum libusb_option opt = LIBUSB_OPTION_NO_DEVICE_DISCOVERY;
-''', dependencies: libusb1_dep, name: 'Has LIBUSB_OPTION_NO_DEVICE_DISCOVERY')
+''', dependencies: libusb_dep, name: 'Has LIBUSB_OPTION_NO_DEVICE_DISCOVERY')
     add_project_arguments('-DHAVE_LIBUSB_OPTION_NO_DEVICE_DISCOVERY1', language: 'c')
   endif
+
+else
+
+  libusb_dep = dependency('libusb', version: libusb_req, required: 'usb' in get_option('iolibs'))
+  if libusb_dep.found()
+    usb0 = true
+    add_project_arguments('-DHAVE_LIBUSB=1', language: 'c')
+  endif
 endif
 
+lockdev_dep = cc.find_library('lockdev', required : false)
+
 if ltdl_dep.found()
```

Comment:
You set it as required just before 😉

Reply:
done.

Reply:
Note that it looks like the autotools check make it optional (it's hard to see) but it seems that the include is unconditional. We can revisit this at a later time. Just that I didn't have libltdl instead and couldn't build.